File: src/main/java/org/railwaystations/rsapi/core/model/ProblemReportType.java
package org.railwaystations.rsapi.core.model;

public enum ProblemReportType {
    WRONG_LOCATION(false),
    STATION_ACTIVE(false),
    STATION_INACTIVE(false),
    STATION_NONEXISTENT(false),
    WRONG_PHOTO(true),
    PHOTO_OUTDATED(true),
    OTHER(false),
    WRONG_NAME(false);

    private final boolean needsPhoto;

    ProblemReportType(boolean needsPhoto) {
        this.needsPhoto = needsPhoto;
    }

    public boolean needsPhoto() {
        return needsPhoto;
    }
}

File: src/main/kotlin/org/railwaystations/rsapi/core/model/ProblemReportType.kt
package org.railwaystations.rsapi.core.model

enum class ProblemReportType(private val needsPhoto: Boolean) {
    WRONG_LOCATION(false),
    STATION_ACTIVE(false),
    STATION_INACTIVE(false),
    STATION_NONEXISTENT(false),
    WRONG_PHOTO(true),
    PHOTO_OUTDATED(true),
    OTHER(false),
    WRONG_NAME(false),
    DUPLICATE(false);

    fun needsPhoto(): Boolean {
        return needsPhoto
    }
}
